src/main/java/com/google/cloud/dataproc/templates/util/DataplexUtil.java

```diff
@@ -0,0 +1,394 @@
+
+  private static String ASSET_ENTITIES_PROP_KEY = "entities";
+  private static String ASSET_ENTITY_NAME_PROP_KEY = "name";
+  private static String ENTITY_BASE_PATH_PROP_KEY = "dataPath";
+  private static String ENTITY_FORMAT_PROP_KEY = "format";
+  private static String ENTITY_SCHEMA_PROP_KEY = "schema";
+  private static String ENTITY_SCHEMA_PARTITION_FIELDS_PROP_KEY = "partitionFields";
+  private static String PARTITION_FIELD_NAME_PROP_KEY = "name";
+  private static String ENTITY_PARTITION_PROP_KEY = "partitions";
+  private static String ENTITY_PARTITION_LOCATION_PROP_KEY = "location";
+  private static String ENTITY_PARTITION_VALUES_PROP_KEY = "values";
+  private static String ENTITY_SCHEMA_FIELD_TYPE_PROP_KEY = "type";
+  private static String ENTITY_SCHEMA_FIELD_MODE_PROP_KEY = "mode";
+  private static String ENTITY_SCHEMA_FIELD_MODE_UNSPECIFIED = "MODE_UNSPECIFIED";
+  private static String ENTITY_SCHEMA_FIELD_MODE_REPEATED = "REPEATED";
+  private static String ENTITY_SCHEMA_TYPE_MODE_RECORD = "RECORD";
+  private static String ENTITY_SCHEMA_FIELDS_PROP_NAME = "fields";
+  private static String DATAPLEX_API_RESP_NEXT_PAGE_TOKEN_FIELD_NAME = "nextPageToken";
+  private static String DATAPLEX_BOOLEAN_DATA_TYPE_NAME = "BOOLEAN";
+  private static String DATAPLEX_BYTE_DATA_TYPE_NAME = "BYTE";
+  private static String DATAPLEX_INT16_DATA_TYPE_NAME = "INT16";
+  private static String DATAPLEX_INT32_DATA_TYPE_NAME = "INT32";
+  private static String DATAPLEX_INT64_DATA_TYPE_NAME = "INT64";
+  private static String DATAPLEX_FLOAT_DATA_TYPE_NAME = "FLOAT";
+  private static String DATAPLEX_DOUBLE_DATA_TYPE_NAME = "DOUBLE";
+  private static String DATAPLEX_DECIMAL_DATA_TYPE_NAME = "DECIMAL";
+  private static String DATAPLEX_STRING_DATA_TYPE_NAME = "STRING";
+  private static String DATAPLEX_BINARY_DATA_TYPE_NAME = "BINARY";
+  private static String DATAPLEX_TIMESTAMP_DATA_TYPE_NAME = "TIMESTAMP";
+  private static String DATAPLEX_DATE_DATA_TYPE_NAME = "DATE";
+
+  /**
+   * Execute request on Google API
+   *
+   * @param url of the API method
+   * @return request response
+   * @throws IOException when request fails
+   */
+  private static JsonObject executeRequest(String url) throws IOException {
+    GoogleCredentials googleCredentials = GoogleCredentials.getApplicationDefault();
+    HttpCredentialsAdapter credentialsAdapter = new HttpCredentialsAdapter(googleCredentials);
+    HttpRequestFactory requestFactory =
+        new NetHttpTransport().createRequestFactory(credentialsAdapter);
+    HttpRequest request = requestFactory.buildGetRequest(new GenericUrl(url));
+    JsonObjectParser parser = new JsonObjectParser(GsonFactory.getDefaultInstance());
+    request.setParser(parser);
+    HttpResponse response = request.execute();
+    String resp = response.parseAsString();
+    return JsonParser.parseString(resp).getAsJsonObject();
+  }
+
+  /**
+   * Execute request on Google API
```

Comment:
nit. Execute request on Google API to get the list of entities in a Dataplex asset